app/src/main/java/com/captech/android_activity_results/MainActivity.kt

```diff
@@ -65,6 +66,19 @@ class MainActivity : AppCompatActivity() {
         )
     }
 
+    /**
+     * Function for onClick from XML
+     */
+    fun onRequestMessage(view: View) {
+        messageContract.launch()
+    }
+
+
+    private val messageContract = registerForActivityResult(MessageContract()) {
+        Toast.makeText(this, it, Toast.LENGTH_SHORT).show()
```

Comment:
Should we add Javadoc comments explaining the use here?

Reply:
ah yes, we shoudl add way more documentation than what I have added :)